Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -103,12 +103,23 @@ JS::GCPtr<Layout::Node> HTMLInputElement::create_layout_node(NonnullRefPtr<CSS::
     if (type_state() == TypeAttributeState::Hidden)
         return nullptr;
 
+    // NOTE: Image inputs are `appearance: none` per the default UA style,
+    //       but we still need to create an ImageBox for them, or no image will get loaded.
+    if (type_state() == TypeAttributeState::ImageButton) {
```

Comment:
Should TypeAttributeState::FileUpload be handled by this PR? It's also specified as `appearance: none` in Default.css

Reply:
Most of the possible type states, like `file`, `text`, `password`, `search`, `color`, `range` etc. are implemented by constructing browser-internal shadow DOM trees with elements that together make up the form control :)

See `HTMLInputElement::create_shadow_tree_if_needed()`